handlers/citations.go

```diff
@@ -0,0 +1,68 @@
+package handlers
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	ml "github.com/KingDerp/memoryLane"
+	"github.com/KingDerp/memoryLane/server"
+
+	"github.com/sirupsen/logrus"
+)
+
+type citationHandler struct {
+	citationServer *server.CitationServer
+}
+
+func newCitationHandler(server *server.CitationServer) *citationHandler {
+	return &citationHandler{citationServer: server}
+}
+
+func (ss *citationHandler) newCitation(w http.ResponseWriter, req *http.Request) {
+
+	ctx := req.Context()
+
+	newCitationRequest, err := unmarshalCitationReq(req.Body)
+	if err != nil {
+		ml.HandleUnmarshalError(w, err)
+		return
+	}
+
+	err = ss.citationServer.NewCitation(ctx, newCitationRequest)
+	if err != nil {
+		ml.HandleCitationError(w, err)
+		return
+	}
+
+	b, err := marshalCitationResp()
+	if err != nil {
+		ml.HandleMarshalError(w, err)
+		return
+	}
+
+	renderCitationResp(w, b)
+}
+
+func renderCitationResp(w http.ResponseWriter, b []byte) {
```

Comment:
I'd consider maybe mutating this to a `RenderJSON` function that takes an `obj interface{}` and does the marshaling as well.